subsys/debug/coredump/coredump_backend_flash_partition.c

```diff
@@ -325,15 +325,15 @@ static void coredump_flash_backend_start(void)
 		 * The header size is rounded up so the beginning of coredump
 		 * is aligned to write size (for easier read and seek).
 		 */
-		offset = backend_ctx.flash_area->fa_off;
-		offset += ROUND_UP(sizeof(struct flash_hdr_t),
-				   FLASH_WRITE_SIZE);
+		header_size = ROUND_UP(sizeof(struct flash_hdr_t),
+				       FLASH_WRITE_SIZE);
```

Comment:
What if you would align this up to FLASH_PAGE_SIZE ? Then you limit the stream flash to a flash_page aligned area and you can treat the header separated from the dump.

Reply:
The issue isn't with the page size for writing, but for erasing.

For example, my flash can write single bytes, however an erase block is 4K.